starknet/src/L1_headers_store/contract.cairo

```diff
@@ -56,7 +76,7 @@ pub mod L1HeaderStore {
         fn initialize(ref self: ContractState, l1_messages_origin: starknet::ContractAddress) {
             assert!(self.initialized.read() == false, "L1HeaderStore: already initialized");
             self.initialized.write(true);
-            self.l1_messages_origin.write(l1_messages_origin);
+            self.ownable.initializer(l1_messages_origin);
```

Comment:
here you can add a param for admin as ContractAddress and init ownable with it, l1_message_sender stays as before